### DISCUSSION TITLE
ERROR: for teslamate  'ContainerConfig' - after updating to v1.28.5

### DISCUSSION THREAD

**Discussion by TThommy** [teslamate-org/teslamate#3776] [Q&A] [answered]
Hello all,

just tried to update my TeslaMate container to v1.28.5. After the pull (sure that worked without any issues), I tried to update the pulled images, as always with the "docker-compose up -d" command. Unfortunately, it ended it an issue, see the following lines.

> pi@raspberrypi:~ $ docker-compose up -d
> pi_database_1 is up-to-date
> Recreating 8fdd3dfa350b_pi_grafana_1 ... 
> Recreating ff2c4bc4e132_pi_teslamate_1 ... 
> pi_mosquitto_1 is up-to-date
> 
> ERROR: for ff2c4bc4e132_pi_teslamate_1  'ContainerConfig'
> 
> ERROR: for 8fdd3dfa350b_pi_grafana_1  'ContainerConfig'
> 
> ERROR: for teslamate  'ContainerConfig'
> 
> ERROR: for grafana  'ContainerConfig'
> Traceback (most recent call last):
>   File "/usr/local/bin/docker-compose", line 8, in <module>
>     sys.exit(main())
>   File "/usr/local/lib/python3.9/dist-packages/compose/cli/main.py", line 81, in main
>     command_func()
>   File "/usr/local/lib/python3.9/dist-packages/compose/cli/main.py", line 203, in perform_command
>     handler(command, command_options)
>   File "/usr/local/lib/python3.9/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
>     result = fn(*args, **kwargs)
>   File "/usr/local/lib/python3.9/dist-packages/compose/cli/main.py", line 1186, in up
>     to_attach = up(False)
>   File "/usr/local/lib/python3.9/dist-packages/compose/cli/main.py", line 1166, in up
>     return self.project.up(
>   File "/usr/local/lib/python3.9/dist-packages/compose/project.py", line 697, in up
>     results, errors = parallel.parallel_execute(
>   File "/usr/local/lib/python3.9/dist-packages/compose/parallel.py", line 108, in parallel_execute
>     raise error_to_reraise
>   File "/usr/local/lib/python3.9/dist-packages/compose/parallel.py", line 206, in producer
>     result = func(obj)
>   File "/usr/local/lib/python3.9/dist-packages/compose/project.py", line 679, in do
>     return service.execute_convergence_plan(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 579, in execute_convergence_plan
>     return self._execute_convergence_recreate(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
>     containers, errors = parallel_execute(
>   File "/usr/local/lib/python3.9/dist-packages/compose/parallel.py", line 108, in parallel_execute
>     raise error_to_reraise
>   File "/usr/local/lib/python3.9/dist-packages/compose/parallel.py", line 206, in producer
>     result = func(obj)
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 494, in recreate
>     return self.recreate_container(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 612, in recreate_container
>     new_container = self.create_container(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 330, in create_container
>     container_options = self._get_container_create_options(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 921, in _get_container_create_options
>     container_options, override_options = self._build_container_volume_options(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 960, in _build_container_volume_options
>     binds, affinity = merge_volume_bindings(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
>     old_volumes, old_mounts = get_container_data_volumes(
>   File "/usr/local/lib/python3.9/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
>     container.image_config['ContainerConfig'].get('Volumes') or {}
> KeyError: 'ContainerConfig'

Any thoughts how to fix this? TeslaMate is not running/accessible via the browser.

Best regards
Thomas

**Comment by eliluong** [comment, marked as answer]
What about running `docker-compose down` first, then recreating the containers? You have to have volumes defined in compose file, otherwise this will delete the volumes - https://dev.to/darkmavis1980/how-to-persist-data-with-docker-compose-ik8

**Reply by TThommy** [reply]
Many thanks for the fast response! The command helped me to get it up and running again. But it’s strange that I never needed to run the command the last >10 updates before.

**Comment by ktsenvolt** [comment]
I am having the same issue since the last 1.28.4 update. I will add "docker-compose down" to my updated command from now on.

**Reply by durzel** [reply]
Had the same issue when updating to this version.

Previously I had a single line to update TeslaMate:

`sudo docker-compose stop ; sudo docker-compose pull ; sudo docker-compose up -d`

..which has always worked, until this version.

I ended up doing `sudo docker-compose down ; sudo docker-compose pull ; sudo docker-compose up -d` which fixed the problem.

Should I be doing `docker-compose down` every time I upgrade?

**Comment by itoffshore** [comment]
the solution to `KeyError: 'ContainerConfig'`:

* use `docker compose` not `docker-compose` with the latest `v2` compose plugin
